File: tugas-9-20251022/cpp/datetime.cpp
#include <iostream>
#include <chrono>
#include <ctime>

int main() {	
	std::string temp = "";
	std::cout << "start time via enter: "; std::getline(std::cin, temp);
	auto start_time = std::chrono::system_clock::now();
	std::cout << "end time via enter: "; std::getline(std::cin, temp);
	auto end_time = std::chrono::system_clock::now();
	
	
	std::time_t st = std::chrono::system_clock::to_time_t(start_time);
	std::time_t et = std::chrono::system_clock::to_time_t(end_time);
	
	std::tm local_sttm = *std::localtime(&st);
	std::tm local_ettm = *std::localtime(&et);
	
	std::cout << "\nYear   : " << 1900 + local_sttm.tm_year << "\n";
    std::cout << "Month  : " << 1 + local_sttm.tm_mon   << "\n";
    std::cout << "Day    : " << local_sttm.tm_mday      << "\n";
    std::cout << "Hour   : " << local_sttm.tm_hour      << "\n";
    std::cout << "Minute : " << local_sttm.tm_min       << "\n";
    std::cout << "Second : " << local_sttm.tm_sec       << "\n";
	
	std::cout << "\nYear   : " << 1900 + local_ettm.tm_year << "\n";
    std::cout << "Month  : " << 1 + local_ettm.tm_mon   << "\n";
    std::cout << "Day    : " << local_ettm.tm_mday      << "\n";
    std::cout << "Hour   : " << local_ettm.tm_hour      << "\n";
    std::cout << "Minute : " << local_ettm.tm_min       << "\n";
    std::cout << "Second : " << local_ettm.tm_sec       << "\n";
	
	std::chrono::duration<double> diff = end_time - start_time;
	std::cout << diff.count() << std::endl;
	return 0;
}

File: tugas-9-20251022/python/date_time.py
from datetime import datetime

def main():
    input("start time via enter")
    start_time = datetime.now()

    input("end time via enter")
    end_time = datetime.now()

    st_year = start_time.year
    st_month = start_time.month
    st_day = start_time.day
    st_hour = start_time.hour
    st_minute = start_time.minute
    st_second = start_time.second

    print("\n")

    et_year = end_time.year
    et_month = end_time.month
    et_day = end_time.day
    et_hour = end_time.hour
    et_minute = end_time.minute
    et_second = end_time.second

    print(st_year, st_month, st_day, st_hour, st_minute, st_second)
    print(et_year, et_month, et_day, et_hour, et_minute, et_second)

    delta_time = end_time - start_time
    print(delta_time.total_seconds())

if __name__ == '__main__': main()